src/process_utility.c

```diff
@@ -1104,13 +1106,19 @@ process_cluster_start(Node *parsetree, ProcessUtilityContext context)
 	Assert(IsA(stmt, ClusterStmt));
 
 	/* If this is a re-cluster on all tables, there is nothing we need to do */
+	/* TODO should we take charge and run our cluster on our tables? */
 	if (NULL == stmt->relation)
 		return false;
 
 	hcache = hypertable_cache_pin();
 	ht = hypertable_cache_get_entry_rv(hcache, stmt->relation);
 
-	if (NULL != ht)
+	if (NULL == ht)
```

Comment:
Now that we're returning `true` in the case where we perform `CLUSTER` there wasn't any benifit from having a unified return at the end of the function instead of returning early when we encounter an unexpected state. Additionally, the late return made it harder for me to figure out what this function was doing when I started reading it.
I left the block and indent because changing it would move a very large amount of variables to the top of the function (I thought I'd seen this done elsewhere in the postgres codebase, though I cannot seem to find it ATM).

Reply:
Two things. 1) we don't use such block. a style thing. 2) a unified return is still useful because of the cleanup that has to be done before the return